### DISCUSSION TITLE
Warning: no matching IRPs found for reply

### DISCUSSION THREAD

**Discussion by uhonermann** [dokan-dev/dokany#1178] [Q&A]
Hello,

we see quite a lot of these warnings in the Windows event log.

Question is, is this something usual we shouldn't care about ?

Regards,
  Uwe

**Comment by Liryna** [comment]
Hi @uhonermann ,

This log is printed when userland (the dokan library and your filesystem) sends a response to the driver while the driver has no trace of the matching IRP (original filesystem request from an external app).
This could happen if:
* Userland is too slow to process the request and the driver previously timeout the original request to avoid possible system deadlock.
* Original IRP was canceled by the external app source and therefore the userland that processed it for "nothing".

It would say it is normal to have the second case if external apps are heavily using cancelation but otherwise there should be none. Are you seeing the same with one of the dokan samples ?

**Comment by uhonermann** [comment]
Hi Liryna,

thank you for your information.

Yesterday I took a much deeper look into the driver and library code and ended up with the same possible reasons.

The problem is, that I couldn't find a way of reproducing this, yet, which also makes it hard to cross-check it using the samples.
I'll add some more debug output (esp. more info about the Events SerialNumber) to figure out why this is happening that often (sometimes).
